### ISSUE TITLE
Heatmap ignoring new startDate and endDate values

### ISSUE THREAD

**Issue by gitname** [patientslikeme/react-calendar-heatmap#59]
When I pass new `startDate` and `endDate` props to a `CalendarHeatmap` component, the component does not update to show the data points that fall within the new date range.

I created an interactive **online demo** in which people can observe this behavior. The demo is at:
https://codesandbox.io/s/32zxwx5ow1

Here's a depiction (created using the above demo) of the behavior:

1. :white_check_mark: This heatmap is configured (via the props first passed to it) to show the entire year of 2017. It, correctly, is showing both "heatmap data" items (since both have a date in 2017).

![image](https://user-images.githubusercontent.com/7143133/33235486-ee8ce3c0-d1ed-11e7-9ebe-7909369be8a6.png)

2. :x: On the other hand, this heatmap is configured (via new props passed to it) to show the entire year of **2018**. However, it continues to show both "heatmap data" items (which both have a date in **2017**, not 2018).

![image](https://user-images.githubusercontent.com/7143133/33235503-3ce88da8-d1ee-11e7-8cab-b95f1ba08155.png)

Pull Request https://github.com/patientslikeme/react-calendar-heatmap/pull/57 contains a test that causes this behavior to occur.

**Comment by gitname** [comment]
In the [online demo linked above](https://codesandbox.io/s/32zxwx5ow1), I noticed that the heatmap shows the two "heatmap data" items in the following cases:
1. When the heatmap is **first rendered** (at which point the active year is 2017) :white_check_mark: 
2. When I change the active year **from 2017** to 2016 :x: 
3. When I change the active year **from 2017** to 2018 :x: 

However, it does not show the two "heatmap data" items in the following cases:
1. When I change the active year **from 2015** to 2016 :white_check_mark: 
2. When I change the active year **from 2019** to 2018 :white_check_mark:

In other words, it the heatmap shows the two "heatmap data" items *when the heatmap is first rendered*, and *the next time it is rendered following when the active year is 2017*.

**Comment by gitname** [comment]
I have my eye on the following code as a potential culprit.

[Lines 19-23 of `index.jsx`](https://github.com/patientslikeme/react-calendar-heatmap/blob/master/src/index.jsx#L21) contain the following:

```js
  componentWillReceiveProps(nextProps) {
    this.setState({
      valueCache: this.getValueCache(nextProps.values),
    });
  }
```

The method `getValueCache()` contains a call to `this.getStartDateWithEmptyDays()`,
the method `getStartDateWithEmptyDays()` contains a call to `this.getStartDate()`,
the method `getStartDate()` contains a call to `this.getEndDate()`,
and [the method `getEndDate()`](https://github.com/patientslikeme/react-calendar-heatmap/blob/master/src/index.jsx#L62-L64) (shown below) contains a reference to `this.props`.

```js
  getEndDate() {
    return getBeginningTimeForDate(convertToDate(this.props.endDate));
  }
```

So, when the component updates its "values cache," it uses `nextProps` to reference the `values`, but uses `this.props` to reference the `endDate`.

**Comment by gitname** [comment]
This issue is fixed in commit 490b665 (part of Pull Request #57).

**Comment by gitname** [comment]
This is fixed in release 1.6.2.

Here is a demo that uses release 1.6.2, showing the absence of the issue:
https://codesandbox.io/s/q7nww60l9j
